applications/sink/rsocket-sink/src/test/java/org/springframework/cloud/stream/app/rsocket/sink/RSocketSinkTests.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2020-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.rsocket.sink;
+
+import org.junit.jupiter.api.Test;
+import reactor.core.publisher.ReplayProcessor;
+import reactor.test.StepVerifier;
+
+import org.springframework.boot.WebApplicationType;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.cloud.fn.consumer.rsocket.RsocketConsumerConfiguration;
+import org.springframework.cloud.stream.binder.test.InputDestination;
+import org.springframework.cloud.stream.binder.test.TestChannelBinderConfiguration;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Import;
+import org.springframework.integration.support.MessageBuilder;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.handler.annotation.MessageMapping;
+import org.springframework.stereotype.Controller;
+
+public class RSocketSinkTests {
+
+	@Test
+	public void testRsocketSink() throws Exception {
+		try (ConfigurableApplicationContext context = new SpringApplicationBuilder(
```

Comment:
This test doesn't finish because you just send a data into an `InputDestination`.
There is a function subscribed to that destination like this `Function<Message<?>, Mono<Void>>`.
And looks like nothing subscribes to that `Mono`.

Probably time to talk to Oleg to see what we can do for this Reactive "consumer" to make a subscription automatic.

In MongoDbConsumer David did a hack like:
```
@Bean
public Consumer<Message<?>> mongodbConsumer(Function<Message<?>, Mono<Void>> mongodbConsumerFunction) {
	return message -> mongodbConsumerFunction.apply(message).subscribe();
}
```

Honestly I'm not a fun of this solution and would prefer to have something automatic on the framework side: some `flatMap()` or `handle()` should be done over there to honor back-pressure.